src/wallet/wallet.cpp

```diff
@@ -3406,6 +3406,13 @@ void CWallet::postInitProcess()
 
 bool CWallet::BackupWallet(const std::string& strDest) const
 {
+    if (m_chain) {
+        WalletBatch batch{GetDatabase()};
+        CBlockLocator loc{m_chain->getTipLocator()};
+        if (!loc.IsNull()) {
+            batch.WriteBestBlock(loc);
+        }
+    }
```

Comment:
If the node's chain is ahead of the wallet, you would be writing a locator in the future. Which would make the wallet skip blocks during the next rescan attempt.
Should lock the wallet mutex and obtain the wallet's best locator from the wallet's last processed block (`m_last_block_processed`).

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30678#discussion_r1722605697

> Should lock the wallet mutex and obtain the wallet's best locator from the wallet's last processed block (`m_last_block_processed`).

Left a similar suggestion https://github.com/bitcoin/bitcoin/issues/30686#issuecomment-2299674756 implementing that